plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -366,7 +368,33 @@ public synchronized void createDatabase(HiveIdentity identity, Database database
 
     public synchronized void dropDatabase(HiveIdentity identity, String schemaName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.dropDatabase(identity, schemaName, true));
+        HdfsContext context = new HdfsContext(
+                identity.getUsername()
+                        .map(ConnectorIdentity::ofUser)
+                        .orElseThrow(() -> new IllegalStateException("username is null")));
+
+        Optional<Path> location = delegate.getDatabase(schemaName)
+                .orElseThrow(() -> new SchemaNotFoundException(schemaName))
+                .getLocation()
+                .map(Path::new);
+
+        setExclusive((delegate, hdfsEnvironment) -> {
+            // If we see files in the schema location, don't delete it.
+            // If we see no files or can't see the location at all, do request deletion.
+            boolean deleteData = location.map(path -> {
+                try {
+                    FileSystem fs = hdfsEnvironment.getFileSystem(context, path);
+                    // If no files in schema directory, delete it
+                    return !fs.listFiles(path, false).hasNext();
+                }
+                catch (IOException e) {
+                    log.warn(e, "Could not check schema directory '%s'", path);
+                    return true;
```

Comment:
Didn't we discuss to have a config property to govern what the default is? I am not sure what we settled to. @findepi ?

Reply:
Oh, oops. I had that, but I forgot to re-add it when I redid the basic logic.